packages/desktop-client/src/components/mobile/transactions/TransactionListWithBalances.tsx

```diff
@@ -0,0 +1,278 @@
+          height: styles.mobileMinHeight,
+        }}
+      />
+    </View>
+  );
+}
+
+type TransactionListWithBalancesProps = {
+  isLoading: boolean;
+  transactions: Readonly<TransactionEntity[]>;
+  balance:
+    | ReturnType<typeof queries.onBudgetAccountBalance>
+    | ReturnType<typeof queries.offBudgetAccountBalance>
+    | ReturnType<typeof queries.uncategorizedBalance>
+    | ReturnType<typeof queries.categoryBalance>
+    | ReturnType<typeof queries.accountBalance>
+    | ReturnType<typeof queries.allAccountBalance>;
+  balanceCleared?:
+    | ReturnType<typeof queries.categoryBalanceCleared>
+    | ReturnType<typeof queries.accountBalanceCleared>;
+  balanceUncleared?:
+    | ReturnType<typeof queries.categoryBalanceUncleared>
+    | ReturnType<typeof queries.accountBalanceUncleared>;
+  searchPlaceholder: string;
+  onSearch: (searchText: string) => void;
+  isLoadingMore: boolean;
+  onLoadMore: () => void;
+  onOpenTransaction: (transaction: TransactionEntity) => void;
+  onRefresh?: () => void;
+};
+
+export function TransactionListWithBalances({
+  isLoading,
+  transactions,
+  balance,
+  balanceCleared,
+  balanceUncleared,
+  searchPlaceholder = 'Search...',
+  onSearch,
+  isLoadingMore,
+  onLoadMore,
+  onOpenTransaction,
+  onRefresh,
+}: TransactionListWithBalancesProps) {
+  const selectedInst = useSelected('transactions', [...transactions], []);
+
+  return (
+    <SelectedProvider instance={selectedInst}>
+      <>
+        <View
+          style={{
+            flexShrink: 0,
+            marginTop: 10,
+          }}
+        >
+          <View
+            style={{
+              flexDirection: 'row',
+              justifyContent: 'space-evenly',
+            }}
+          >
+            {balanceCleared && balanceUncleared ? (
+              <BalanceWithCleared
+                balance={balance}
+                balanceCleared={balanceCleared}
+                balanceUncleared={balanceUncleared}
+              />
+            ) : (
+              <Balance balance={balance} />
+            )}
+          </View>
+          <TransactionSearchInput
+            placeholder={searchPlaceholder}
+            onSearch={onSearch}
+          />
+        </View>
+        <PullToRefresh
+          isPullable={!!onRefresh}
+          onRefresh={async () => onRefresh?.()}
+        >
+          <TransactionList
+            isLoading={isLoading}
+            transactions={transactions}
+            isLoadingMore={isLoadingMore}
+            onLoadMore={onLoadMore}
+            onOpenTransaction={onOpenTransaction}
+          />
+        </PullToRefresh>
+      </>
+    </SelectedProvider>
+  );
+}
+
+const TransactionListBalanceCellValue = <
+  FieldName extends SheetFields<'account'> | SheetFields<'category'>,
+>(
+  props: ComponentProps<
+    typeof CellValue<
+      FieldName extends SheetFields<'account'> ? 'account' : 'category',
+      FieldName
+    >
+  >,
+) => {
+  return <CellValue {...props} />;
+};
```

Comment:
Following the pattern found in `budget/envelope/EnvelopeBudgetCompontns.tsx` and `budget/tracking/TrackingBudgetCompontns.tsx`